### DISCUSSION TITLE
best practices

### DISCUSSION THREAD

**Discussion by pankkovv** [software-sales-and-installations/softplat-back#2] [Ideas]
активайция среды разработки - https://blog.llinh9ra.ru/софт/активация-phpstorm-webstorm-intellij-idea-и-другие-продукты-jetbrains-в/
mapstruct - https://javarush.com/groups/posts/3698-chto-takoe-mapstruct-i-kak-praviljhno-nastroitjh-ego-dlja-moduljhnogo-testirovanija-v-springboo
внедрение Spring Security - ...

**Comment by Wereflyers** [comment]
Тесты Security
https://www.baeldung.com/spring-security-integration-tests

**Comment by Wereflyers** [comment]
QueryDSL
https://www.youtube.com/watch?v=gBK2lar-Z7E&t=412s

**Comment by Wereflyers** [comment]
Swagger
https://struchkov.dev/blog/ru/api-swagger/

**Comment by mdemidkin1992** [comment]
**Swagger** 
- Настройка spring-fox (https://springfox.github.io/springfox/docs/current/#configuring-springfox)
- Основные аннотации для контроллеров (https://springfox.github.io/springfox/docs/current/#property-file-lookup)

**Comment by pankkovv** [comment]
Теория о правильной индексации записей в SQL:
https://use-the-index-luke.com/

**Comment by mdemidkin1992** [comment]
Вводная в мейлинг для Спринг Бута 
https://www.baeldung.com/spring-email
